src/engine.go

```diff
@@ -159,6 +167,71 @@ func (e *engine) render() {
 	e.write()
 }
 
+// debug will lool through your config file and output the timings for each segments
+func (e *engine) debug() {
+	var segmentTimings []SegmentTiming
+	nameMaxLength := 0
+	e.renderer.print("\nHere are the timings of segments in your prompt:\n")
+	// loop each segments of each blocks
+	for _, block := range e.settings.Blocks {
+		for _, segment := range block.Segments {
+			err := segment.mapSegmentWithWriter(e.env)
+			if err != nil || segment.shouldIgnoreFolder(e.env.getcwd()) {
+				return
+			}
+
+			var segmentTiming SegmentTiming
+
+			segmentTiming.name = string(segment.Type)
+			segmentTiming.nameLength = len(segmentTiming.name)
+
+			if segmentTiming.nameLength > nameMaxLength {
+				nameMaxLength = segmentTiming.nameLength
+			}
+
+			segmentTiming.background = segment.Background
+			segmentTiming.foreground = segment.Foreground
+
+			// enabled timing
+			start := time.Now()
+			segmentTiming.enabled = segment.enabled()
+			segmentTiming.enabledDuration = time.Since(start)
+
+			// string timing
+			if segmentTiming.enabled {
+				start = time.Now()
+				segmentTiming.stringValue = segment.string()
+				segmentTiming.stringDuration = time.Since(start)
+
+				// not pretty rendering could be refactored for a better separation of concern
+				e.previousActiveSegment = nil
+				e.activeSegment = segment
+				e.activeSegment.Background = segment.props.background
+				e.activeSegment.Foreground = segment.props.foreground
+				e.renderSegmentText(segmentTiming.stringValue)
+				segmentTiming.stringValue = e.color.string()
+				e.color.buffer.Reset()
+			}
+
+			segmentTimings = append(segmentTimings, segmentTiming)
+		}
+	}
+
+	// 7 => (false)
+	nameMaxLength += 7
```

Comment:
what's the meaning of this?

Reply:
To align the columns, I get the length of the longest segment names + 7 char, corresponding to (false).